### DISCUSSION TITLE
Support for separate scenes/sub-projects

### DISCUSSION THREAD

**Discussion by idoblenderstuffs** [ScratchAddons/ScratchAddons#6041] [Ideas]
This is a feature I actually thought of due to ScratchJr. In that, you can split your project into separate scenes or sub-projects. It makes organising your very large projects a lot easier and most game engines have something like this. Obviously, this is a monumental task. I can kinda think some some dodgey ways to implement something like it, but it’s honestly probably up to Scratch Team themselves to do this in Scratch 4.0 or something. If anyone here thinks they could still maybe accomplish this with Scratch Addons or maybe the implementation in TurboWarp then that would be REALLY cool. I’m currently working on a large project in TurboWarp which is split up into separate chapters with very different gameplay, so I thought this feature would be a nice addition. I know we do have the sprite folders addon (THANK YOU) which will suffice for me, but still, scenes would be even better. I’ve already suggested this on the actual Scratch forums too as a real feature, but you know what ST is like.

**Comment by BroJac5246** [comment]
How would this work, particularly when a project made with it is used by someone who doesn't use Scratch Addons?

**Reply by idoblenderstuffs** [reply]
I’m pretty sure there’s some Scratch Addons which are literally just labelled as not compatable with Scratch, so it’ll likely be one of those.

**Reply by BroJac5246** [reply]
> I’m pretty sure there’s some Scratch Addons which are literally just labelled as not compatable with Scratch, so it’ll likely be one of those.

Not really. Some features (for example the debugger) don't work with Scratch, but are designed not to change the experience in vanilla Scratch (in the debugger's case, the blocks are just custom blocks that don't do anything in vanilla Scratch). Scratch Addons explicitly does not allow any addons that break functionally for vanilla Scratch users, and all projects created with Scratch Addons must work the same way with Scratch Addons disabled. Also note that some features (for example the ability to use more keys in the blocks that detect keep presses) appear like they won't work in vanilla Scratch but are actually just hidden Scratch features.

**Reply by idoblenderstuffs** [reply]
oh right. i must’ve been thinking of the other weird addons that turbowarp lets you use.
